diff --git a/config/config.go b/config/config.go
index 16364be..1249f73 100644
--- a/config/config.go
+++ b/config/config.go
@@ -131,6 +131,9 @@ type SystemConfiguration struct {
 	// Directory where the server data is stored at.
 	Data string `default:"/var/lib/pelican/volumes" json:"-" yaml:"data"`
 
+	//Directory where the SFTP Data is stored at. Please keep this the same as the data entry
+	SFTPData string `default:"/var/lib/pelican/volumes" json:"-" yaml:"sftp_data"`
+
 	// Directory where server archives for transferring will be stored.
 	ArchiveDirectory string `default:"/var/lib/pelican/archives" json:"-" yaml:"archive_directory"`
 
diff --git a/server/manager.go b/server/manager.go
index 55dd645..520efb6 100644
--- a/server/manager.go
+++ b/server/manager.go
@@ -196,7 +196,12 @@ func (m *Manager) InitServer(data remote.ServerConfigurationResponse) (*Server,
 		return nil, errors.WithStackIf(err)
 	}
 
-	s.fs, err = filesystem.New(filepath.Join(config.Get().System.Data, s.ID()), s.DiskSpace(), s.Config().Egg.FileDenylist)
+	s.fs, err = filesystem.New(filepath.Join(config.Get().System.Data, s.ID())+"/data", s.DiskSpace(), s.Config().Egg.FileDenylist)
+	if err != nil {
+		return nil, errors.WithStackIf(err)
+	}
+
+	s.sftpFs, err = filesystem.New(filepath.Join(config.Get().System.Data, s.ID()), s.DiskSpace(), s.Config().Egg.FileDenylist)
 	if err != nil {
 		return nil, errors.WithStackIf(err)
 	}
diff --git a/server/server.go b/server/server.go
index d4ab599..31b2b50 100644
--- a/server/server.go
+++ b/server/server.go
@@ -48,7 +48,8 @@ type Server struct {
 	resources   ResourceUsage
 	Environment environment.ProcessEnvironment `json:"-"`
 
-	fs *filesystem.Filesystem
+	fs     *filesystem.Filesystem
+	sftpFs *filesystem.Filesystem
 
 	// Events emitted by the server instance.
 	emitter *events.Bus
@@ -434,6 +435,10 @@ func (s *Server) ToAPIResponse() APIResponse {
 	}
 }
 
+func (s *Server) SFTPFilesystem() *filesystem.Filesystem {
+	return s.sftpFs
+}
+
 func (s *Server) RemoveAllServerBackups() error {
 	sp := path.Join(config.Get().System.BackupDirectory, s.ID())
 	// This should never be possible, but we'll check it anyway.
diff --git a/sftp/handler.go b/sftp/handler.go
index dacea40..f49f8d0 100644
--- a/sftp/handler.go
+++ b/sftp/handler.go
@@ -52,7 +52,7 @@ func NewHandler(sc *ssh.ServerConn, srv *server.Server) (*Handler, error) {
 	return &Handler{
 		permissions: strings.Split(sc.Permissions.Extensions["permissions"], ","),
 		server:      srv,
-		fs:          srv.Filesystem(),
+		fs:          srv.SFTPFilesystem(),
 		events:      &events,
 		ro:          config.Get().System.Sftp.ReadOnly,
 		logger:      log.WithFields(log.Fields{"subsystem": "sftp", "user": uuid, "ip": sc.RemoteAddr()}),
